beacon-chain/db/kv/state.go

```diff
@@ -418,6 +418,9 @@ func (s *Store) validatorEntries(ctx context.Context, blockRoot [32]byte) ([]*v1
 				}
 				validatorEntries = append(validatorEntries, encValEntry)
 				validatorEntryCacheMiss.Inc()
+
+				// should add here in cache
+				s.validatorEntryCache.Set(key, encValEntry, int64(encValEntry.SizeSSZ()))
```

Comment:
Is there any concern with cardinality? How often does this happen?

Reply:
Nope. Updates to the entries of this bucket are low (per spec).
Most of the times they are the same, per my analysis of the mainnet DB and conformed by the cache hit count in the screen shot attached.